### PR TITLE
DBZ-146 Improved error handling of MySQL Connector

### PR DESCRIPTION
**Pull request by rhauch**
Improved the error handling of the MySQL connector to ensure that we’re always stopping the connector when we have a problem handling a binlog event or if we have problems starting.

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java [hunk 1]
```diff
@@ -387,6 +386,15 @@ protected void handleGtidEvent(Event event) {
     protected void handleQueryEvent(Event event) {
         QueryEventData command = unwrapData(event);
         logger.debug("Received update table command: {}", event);
+        String sql = command.getSql().trim();
+        if ( sql.equalsIgnoreCase("BEGIN") ) {
```
Comment (criccomini) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java]: Nit: space

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java [hunk 2]
```diff
@@ -296,6 +287,14 @@ protected void handleEvent(Event event) {
 
             // And after that event has been processed, always set the starting row number to 0 ...
             startingRowNumber = 0;
+        } catch (RuntimeException e) {
+            // There was an error in the event handler, so propagate the failure to Kafka Connect ...
+            failed(e,"Error processing binlog event");
```
Comment (criccomini) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java]: Nit: space

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java [hunk 3]
```diff
@@ -171,10 +174,21 @@ public void start(Map<String, String> props) {
...
-            this.taskContext.shutdown();
-            throw e;
+        } catch (Throwable e) {
+            // If we don't complete startup, then Kafka Connect will not attempt to stop the connector. So if we
+            // run into a problem, we have to stop ourselves ...
+            try {
+                stop();
+            } catch (Throwable s) {
+                // Log, but don't propagate ...
+                logger.error("Failed to start the connector (see other exception), but got this error while cleaning up",s);
```
Comment (criccomini) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java]: Nit: space

**Comment by rhauch** [comment]
Rebased and corrected the formatting issues.

**Pull request merged**